### DISCUSSION TITLE
Cannot curl service from different node

### DISCUSSION THREAD

**Discussion by luoxiaopeng** [k3s-io/k3s#8093] [General]
I create 4 nginx service, show
kubectl get pods -o wide (hide some unimportant)
test-nginx     1/1      Running   10.42.1.3   k8s-worker1
test-nginx2   1/1     Running   10.42.0.9   k8s-master

Then i curl 10.42.1.3 on k8s-master, no response.
But i can curl 10.42.1.9 on k8s-master
Also,i can't curl 10.42.1.9 on k8s-worker1

k8s-master is Ubuntu 20.04.4 LTS with iptables v1.8.4 (legacy)
k8s-worker1 is Ubuntu 22.04.1 LTS with iptables v1.8.7 (nf_tables) (maybe this reason)

Did I miss something?

**Comment by luoxiaopeng** [comment]
I have resolved this problem, just add a param --flannel-external-ip when install k3s, because of k8s-master and k8s-worker1 on different vendor vps。

**Comment by dillfrescott** [comment]
Having the same issues. I'm about to give up on k3s
